### ISSUE TITLE
One behavior diff with encoding/json

### ISSUE THREAD

**Issue by RainshawGao** [bytedance/sonic#251]
Hello, I'm trying to add sonic support in [gin](https://github.com/gin-gonic/gin/pull/3184) . And I have found that sonic has a different behavior with encoding/json.

when try to use newDecoder().decode func on an empty string, encoding/json would return an EOF error, while sonic return nothing and an empty struct.

As addressed in https://stackoverflow.com/questions/18419428/what-is-the-minimum-valid-json , an empty string is not a valid json string, so could you please make this behavior the same with encoding/json?

**Comment by chenzhuoyu** [comment]
@RainshawGao Could you please provide a minimal repro?

**Comment by RainshawGao** [comment]
sure!

```golang
package main

import (
	"bytes"
	"encoding/json"
	"fmt"

	"github.com/bytedance/sonic"

	jsoniter "github.com/json-iterator/go"
)

func main() {
	type test struct {
		Name string `json:"name"`
	}
	var data = ``
	test1 := &test{}
	err := json.NewDecoder(bytes.NewBufferString(data)).Decode(test1)
	if err != nil {
		fmt.Printf("std json err: %v\n", err)
	} else {
		fmt.Printf("std json ok: %v\n", test1)
	}

	err = jsoniter.ConfigCompatibleWithStandardLibrary.NewDecoder(bytes.NewBufferString(data)).Decode(test1)
	if err != nil {
		fmt.Printf("josniter err: %v\n", err)
	} else {
		fmt.Printf("josniter ok\n")
	}

	err = sonic.ConfigStd.NewDecoder(bytes.NewBufferString(data)).Decode(test1)
	if err != nil {
		fmt.Printf("sonic err: %v\n", err)
	} else {
		fmt.Printf("sonic ok\n")
	}
}
```

run this gives 
```
std json err: EOF
josniter err: EOF
sonic ok
```

**Comment by chenzhuoyu** [comment]
@RainshawGao Thanks, we will investigate on this.

**Comment by chenzhuoyu** [comment]
@RainshawGao Solved in #254.
